Corvus.Extensions.Samples/TraversalExtensionsSample.dib

```diff
@@ -0,0 +1,251 @@
+#!markdown
+
+## Traversal Extensions - Sample Notebook
+
+#!markdown
+
+First, add a reference to the `Corvus.Extensions` NuGet package.
+
+#!csharp
+
+#r "nuget: Corvus.Extensions, 1.1.4"
+
+using Corvus.Extensions;
+
+#!markdown
+
+### ForEachAsync()
+
+#!markdown
+
+Execute an async action for each item in the enumerable.
+
+#!csharp
+
+public async Task<bool> isEqualToOne(int x)
+{
+    bool result = x.Equals(1);
+    Console.WriteLine(result);
+    return result;
+}
+
+Func<int, Task<bool>> funcIsEqualToOne = isEqualToOne;
+
+#!csharp
+
+var list = new List<int> { 1, 2, 3, 4, 5 };
+await list.ForEachAsync<int>(isEqualToOne)
+
+#!markdown
+
+The `ForEachAsync()` method is here for historical interest. It is now possible to write the following and get the same output:
+
+#!csharp
+
+foreach (int x in list)
+{
+    await isEqualToOne(x);
+}
+
+#!markdown
+
+### ForEachAtIndex<T>()
+
+#!markdown
+
+Execute an action for each item in the enumerable with the index of the item in the enumerable.
+
+#!csharp
+
+static void ConsolePrint(string s, int i)
+{
+    Console.WriteLine(i);
+    Console.WriteLine(s);
+}
+
+Action<string, int> printAction = ConsolePrint;
+
+#!csharp
+
+IEnumerable<string> list = new List<string> { "foo", "bar", "quux" };
+list.ForEachAtIndex<string>(printAction);
+
+#!csharp
+
+int i = 0;
+
+foreach (string v in list)
+{
+    printAction(v, i++);
+}
+
+#!markdown
+
+### ForEachAtIndexAsync()
+
+#!markdown
+
+Execute an async action for each item in the enumerable, in turn, with the index of the item in the enumerable.
+
+#!csharp
+
+static async Task<string> ConsolePrint(string s, int i)
```

Comment:
Again, I think returning a `Task<string>` instead of just a `Task` from this method is potentially confusing since the `ForEachAtIndexAsync` method won't do anything with the resulting `string`.